### PR TITLE
changes to read zendesk.yml and scalastyle configuration

### PR DESCRIPTION
**Pull request by vats1989**
- changes to read Yaml file fron zendesk_configuration library.
- Also added scalastyle checking

### CODE REVIEW

File: src/test/scala/com/zendesk/gatling/configuration/Input.scala [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (haleystorm) [inline src/test/scala/com/zendesk/gatling/configuration/Input.scala]: scala style - typically no new line before package declaration. might be able to get the scala style config to check for this.

File: src/test/scala/com/zendesk/gatling/defaults/Defaults.scala [hunk 2]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (haleystorm) [inline src/test/scala/com/zendesk/gatling/defaults/Defaults.scala]: maybe your editor is adding these new lines?

File: src/test/scala/com/zendesk/gatling/configuration/ZendeskYml.scala [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+case class SystemUserAuth(
+  mac_key: String,
+  gooddata: Map[String, String],
+  zendesk: Map[String, String],
+  zendesk_nps: Map[String, String],
+  certificates: Map[String, String],
+  inbox: Map[String, String],
+  help_center: Map[String, String]
+)
```
Comment (anotherminh) [inline src/test/scala/com/zendesk/gatling/configuration/ZendeskYml.scala]: missing newline

File: src/test/scala/com/zendesk/gatling/configuration/Input.scala [hunk 4]
```diff
@@ -9,29 +10,42 @@ import scala.io.Source
...
+  lazy val input_file_name: String = sys.env.get("input").mkString
 
   private lazy val input_file: String = s"/data/$input_file_name.json"
 
-  val loadInputData: JsValue = Json.parse(getFileData(input_file))
+  lazy val loadInputData: JsValue = Json.parse(getFileData(input_file))
 
-  lazy val test_env: String =
-    if ((loadInputData \ "test_env").asOpt[String].isEmpty)
+  lazy val test_env: String = {
```
Comment (anotherminh) [inline src/test/scala/com/zendesk/gatling/configuration/Input.scala]: Should.. all be camel case?

File: src/test/scala/com/zendesk/gatling/configuration/Input.scala (hunk 1 above)
Comment (vats1989) [reply src/test/scala/com/zendesk/gatling/configuration/Input.scala]: no actually i added scalastyle so it was complaining. So i added new line before package. After adding new line, warning disappeared.

File: src/test/scala/com/zendesk/gatling/defaults/Defaults.scala (hunk 2 above)
Comment (vats1989) [reply src/test/scala/com/zendesk/gatling/defaults/Defaults.scala]: No i added new line to all file.

File: src/test/scala/com/zendesk/gatling/configuration/Input.scala (hunk 1 above)
Comment (vats1989) [reply src/test/scala/com/zendesk/gatling/configuration/Input.scala]: I just changed HeaderMatchesChecker to false in scalastyle-config.xml and removed new line before package and now its not complaining. So will change this to every file.

File: src/test/scala/com/zendesk/gatling/configuration/ZendeskYml.scala [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+object ZendeskYml {
+  implicit val formats = native.Serialization.formats(NoTypeHints)
+
+  private lazy val yaml = loadFromFile("zendesk.yml").extractObject[Map[String, JObject]](test_env)
+  lazy val system_user_auth = yaml.get("system_user_auth").map(_.extract[SystemUserAuth])
+
+  lazy val OAUTH_KEY = system_user_auth.get.mac_key
+  lazy val OAUTH_SECRET = system_user_auth.get.zendesk.get("oauth").mkString
+  lazy val OAUTH_ACCOUNT_CREATION = yaml.get("account_creation_tokens").map(_.extract[AccountCreationTokens]).get.qa
+  /*private def loadFromFile(fileName: String): YamlDoc = {
```
Comment (anotherminh) [inline src/test/scala/com/zendesk/gatling/configuration/ZendeskYml.scala]: Why commented out? Can they be removed?

File: src/test/scala/com/zendesk/gatling/helpers/AccountHelper.scala [hunk 6]
```diff
@@ -1,18 +1,18 @@
...
-import com.zendesk.gatling.configuration.{Input, Application}
+import com.zendesk.gatling.configuration.{Input, ZendeskYml}
 import scalaj.http.Http
 import scala.util.Random
 
 object AccountHelper {
 
-  lazy val authHeader: String = "Bearer ".concat(Application.OAUTH_ACCOUNT_CREATION)
+  lazy val authHeader: String = "Bearer ".concat(ZendeskYml.OAUTH_ACCOUNT_CREATION)
   lazy val pod_id = Input.pod_id
```
Comment (anotherminh) [inline src/test/scala/com/zendesk/gatling/helpers/AccountHelper.scala]: should be camel case

File: src/test/scala/com/zendesk/gatling/configuration/ZendeskYml.scala (hunk 3 above)
Comment (anotherminh) [reply src/test/scala/com/zendesk/gatling/configuration/ZendeskYml.scala]: should be camel case. If it breaks json serialization you can specify `snakify` on the json parser I think

File: src/test/scala/com/zendesk/gatling/configuration/ZendeskYml.scala (hunk 5 above)
Comment (vats1989) [reply src/test/scala/com/zendesk/gatling/configuration/ZendeskYml.scala]: I just want to keep there for my reference. I will remove in next commit.

File: src/test/scala/com/zendesk/gatling/helpers/AccountHelper.scala (hunk 6 above)
Comment (vats1989) [reply src/test/scala/com/zendesk/gatling/helpers/AccountHelper.scala]: I will change all variable to camel case in next PR.

**Pull request merged**